impeller/renderer/backend/metal/command_buffer_mtl.mm

```diff
@@ -171,6 +176,53 @@ static bool LogMTLCommandBufferErrorIfPresent(id<MTLCommandBuffer> buffer) {
   return true;
 }
 
+bool CommandBufferMTL::SubmitCommandsAsync(
+    std::shared_ptr<RenderPass> render_pass) {
+  TRACE_EVENT0("impeller", "CommandBufferMTL::SubmitCommandsAsync");
+  if (!IsValid() || !render_pass->IsValid()) {
+    return false;
+  }
+  auto context = context_.lock();
```

Comment:
I think this is more or less what ended up being the fix for problems around ownership in Vulkan backend.